xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -757,7 +746,7 @@ CGUIFontTTF::Character* CGUIFontTTF::GetCharacter(character_t chr, FT_UInt glyph
   // perform binary search on sorted array by m_glyphAndStyle and
   // if not found obtains position to insert the new m_char to keep sorted
   int low = 0;
-  int high = m_numChars - 1;
+  int high = m_char.size() - 1;
```

Comment:
size is only updated when resize() is used. reserve basically does not cause size() to change.
```
#include <iostream>
#include <vector>

int main()
{
  std::vector<int> vec;
  std::cout << vec.size() << std::endl; //0
  vec.reserve(100);
  std::cout << vec.size() << std::endl; //0
  vec.push_back(1);
  std::cout << vec.size() << std::endl; //1

  return 0;
}
```

Reply:
Yes, I know. Here in the first call size() is zero and while loop is not executed.

Next is reserved capacity of 64 characters (memory allocated only, size stills zero)
```c++
  // increase the size of the buffer if we need it
  if (m_char.size() == m_char.capacity())
  {
    m_char.reserve(m_char.capacity() + CHAR_CHUNK);
    startIndex = 0;
  }
```

and here size() is 1 in the first call and increased +1 next calls

` m_char.emplace(m_char.begin() + low);`

When size() reaches 64 in previous block (64 == 64) capacity is increased to 128....

and size is 65, 66, etc.   

in size() == 128 is reserved capacity of 192...

Reply:
Great! Thanks. Depending on the average value we get, a bigger first reserve might be beneficial. Might...

Reply:
Yes and no. 27 different instances of `GUIFontTTF` are used, one for each font size. It's a very big mess... Some instances are used a lot and reach 170 - 180 glyphs, others (large font sizes) only 20 - 30 glyphs or grow very slowly over time.

If it were a single instance, it would be best to initialize with a 512 buffer and forget about it. Probably 64 is a good number that "works" for all instances.

Some instances are never used (it will also depend on the skin) and the ideal would be not to allocate any memory until it is needed... BUT this code forces that at initialization at least one character is stored for the 27 font sizes:

https://github.com/xbmc/xbmc/blob/3c71103d156f48f092de8c2b6832b7601abd289a/xbmc/guilib/GUIFontTTF.cpp#L320-L323